tests/apps/flask_app.py

```diff
@@ -3,77 +3,100 @@
 from flask.views import MethodView
 
 import marshmallow as ma
-from webargs import fields
+from webargs import fields, dict2schema
 from webargs.flaskparser import parser, use_args, use_kwargs
 from webargs.core import MARSHMALLOW_VERSION_INFO
 
+if MARSHMALLOW_VERSION_INFO[0] < 3:
+    schema_kwargs = {"strict": True}
+else:
+    schema_kwargs = {"unknown": ma.EXCLUDE}
```

Comment:
Is this necessary? Seems like the tests shouldn't be sending any unexpected keys.

Reply:
There's currently a test which explicitly checks that an unexpected key is ignored -- behavior which shouldn't be the norm going forward. So for now this is just duct-tape to make that test work.

I think we want tests, probably with `skipif`s for marshmallow 2, which explores the various `unknown` settings you could pass to the schema, to replace this.

I'll make that change a priority, so that we can talk about whether or not we like those tests.

Reply:
05d308e changes this around a bit.
The `unknown=EXCLUDE` is now used (1) in an explicit "ignores extra" route and (2) in `/echo_headers`, to handle the presence of various headers not explicitly set by the client.